torch/_inductor/codegen/cpp.py

```diff
@@ -1231,6 +1231,30 @@ def bitwise_left_shift(a, b):
     def bitwise_right_shift(a, b):
         return f"{a} >> {b}"
 
+    @staticmethod
+    def remainder(a, b):
+        # Since we register remainder as ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT in
+        # the lowering phase: https://github.com/pytorch/pytorch/blob/
+        # eb1583dbc1a8c4a8f0d1525438f16350f11f1208/torch/_inductor/lowering.py#L5853
+        # Should expect same dtype for a and b. Otherwise, we need to handle
+        # the case when a and b with mixed integer and float.
+        assert (
+            a.dtype == b.dtype
+        ), "remainder vec implementation expect the same inputs' dtype."
+        if is_float_dtype(a.dtype):
+            return f"{a} - ({a} / {b}).floor() * {b}"
+        else:
+            assert is_integer_dtype(a.dtype)
+            if a.dtype in [torch.uint8, torch.int8]:
+                # Since we load 16 int8 elements into vec512, the remaining bits of vec512 could be zero
+                # and causes div with float exception. Convert to int for div to avoid this issue.
```

Comment:
Do we have similar issue for `floordiv` and `truncdiv`?

Reply:
Thanks for the comment.

- Yes, `floordiv` has the same issue and move this part of code into the implementation of `floordiv`
- `truediv` doesn't have this issue since the data type has been casted to `float` in the lowering phase https://github.com/pytorch/pytorch/blob/d0ad13fa42fc2e9935bd3bda2937a3491276d274/torch/_inductor/lowering.py#L5503-L5512